modules/apps/configuration-admin/configuration-admin-test/src/testIntegration/java/com/liferay/configuration/admin/internal/search/test/BaseConfigurationModelTest.java

```diff
@@ -0,0 +1,377 @@
+				ExtendedObjectClassDefinition.Scope.SYSTEM.toString());
+
+		ExtendedObjectClassDefinition extendedObjectClassDefinition =
+			new BaseConfigurationModelTest.SimpleExtendedObjectClassDefinition(
+				extensionAttributes, pid);
+
+		Object configurationModel = configurationModelConstructor.newInstance(
+			extendedObjectClassDefinition, configuration,
+			bundle.getSymbolicName(), StringPool.QUESTION, false);
+
+		Document document = indexer.getDocument(configurationModel);
+
+		documents.add(document);
+
+		indexWriterHelper.addDocument(
+			indexer.getSearchEngineId(), CompanyConstants.SYSTEM, document,
+			true);
+
+		return configurationModel;
+	}
+
+	protected void assertSearchResults(String id) throws Exception {
+		Hits hits = search(id);
+
+		Assert.assertEquals(hits.toString(), 1, hits.getLength());
+	}
+
+	protected void assertSearchNoResults(String id) throws Exception {
+		Hits hits = search(id);
+
+		Assert.assertEquals(hits.toString(), 0, hits.getLength());
+	}
+
+	protected Hits search(String id) throws SearchException {
+		SearchContext searchContext = new SearchContext();
+
+		searchContext.setAndSearch(false);
+		searchContext.setCompanyId(CompanyConstants.SYSTEM);
+		searchContext.setKeywords(id);
+		searchContext.setLocale(LocaleUtil.US);
+
+		return indexer.search(searchContext);
+	}
+
+	protected Bundle getBundle(String bundleSymbolicName) {
+		Bundle[] bundles = bundleContext.getBundles();
+
+		for (Bundle bundle : bundles) {
+			if (bundleSymbolicName.equals(bundle.getSymbolicName())) {
+				return bundle;
+			}
+		}
+
+		return null;
+	}
+
+	protected String getFactoryPid(Object configurationModel)
+		throws IllegalAccessException, InvocationTargetException,
+			   NoSuchMethodException {
+
+		Class<?> configurationModelClass = configurationModel.getClass();
+
+		Method getFactoryPid = configurationModelClass.getMethod(
+			"getFactoryPid");
+
+		Object invoke = getFactoryPid.invoke(configurationModel);
+
+		return invoke.toString();
+	}
+
+	protected String getId(Object configurationModel)
+		throws IllegalAccessException, InvocationTargetException,
+			   NoSuchMethodException {
+
+		Class<?> configurationModelClass = configurationModel.getClass();
+
+		Method getID = configurationModelClass.getMethod("getID");
+
+		Object invoke = getID.invoke(configurationModel);
+
+		return invoke.toString();
+	}
+
+	protected static final String PID =
+		"com.liferay.configuration.admin.internal.search.test." +
+			"BaseConfigurationModelTest";
+
+	protected Bundle bundle;
+	protected BundleContext bundleContext;
+	protected Constructor configurationModelConstructor;
+	protected final List<Document> documents = new ArrayList<>();
+
+	@Inject(filter = "component.name=*.ConfigurationModelIndexer")
+	protected Indexer indexer;
+
+	@Inject
+	protected IndexWriterHelper indexWriterHelper;
+
+	protected Locale defaultLocale;
+
+	protected class SimpleExtendedAttributeDefinition
```

Comment:
If we convert this class to a fixture, could we make these classes top-level, instead of nested?